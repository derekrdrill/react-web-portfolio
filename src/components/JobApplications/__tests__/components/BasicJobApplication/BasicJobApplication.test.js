import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import {
  BasicJobApplication,
  JobAppContainer,
  JobAppTitleText,
  FileUploadContainer,
  FileUploadScrollContainer,
  FileUpload,
} from '../../../components/BasicJobApplication/BasicJobApplication';

configure({ adapter: new Adapter() });

const renderer = new ShallowRenderer();

describe('Basic Job Application tests', () => {
  let realUseContext;
  let useContextMock;

  beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
  });

  afterEach(() => {
    React.useContext = realUseContext;
  });

  it('renders correctly', () => {
    useContextMock.mockReturnValue({ darkMode: true });
    renderer.render(<BasicJobApplication />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('renders JobAppContainer styled component correctly', () => {
    const jobAppContainer = shallow(<JobAppContainer />);
    const jobAppContainerDark = shallow(<JobAppContainer darkMode />);

    expect(jobAppContainer.props().className).toEqual('sc-iBkjds SzBrj');
    expect(jobAppContainerDark.props().className).toEqual('sc-iBkjds kFPvdU');
  });

  it('renders JobAppTitleText styled component correctly', () => {
    const jobAppTitleText = shallow(<JobAppTitleText />);
    const jobAppTitleTextDark = shallow(<JobAppTitleText darkMode />);

    expect(jobAppTitleText.props().className).toEqual('sc-papXJ gFZmeb');
    expect(jobAppTitleTextDark.props().className).toEqual('sc-papXJ gWbQGy');
  });

  it('renders FileUploadContainer styled component correctly', () => {
    const fileUploadContainer = shallow(<FileUploadContainer />);
    const fileUploadContainerDark = shallow(<FileUploadContainer darkMode />);

    expect(fileUploadContainer.props().className).toEqual('sc-kDDrLX ePjEtQ');
    expect(fileUploadContainerDark.props().className).toEqual('sc-kDDrLX bcGSeG');
  });

  it('renders FileUploadScrollContainer styled component correctly', () => {
    const fileUploadScrollContainer = shallow(<FileUploadScrollContainer />);
    const fileUploadScrollContainerDark = shallow(<FileUploadScrollContainer darkMode />);

    expect(fileUploadScrollContainer.props().className).toEqual('sc-iqcoie bjyZvR');
    expect(fileUploadScrollContainerDark.props().className).toEqual('sc-iqcoie iYhDTJ');
  });

  it('renders FileUpload styled component correctly', () => {
    const fileUpload = shallow(<FileUpload />);
    const fileUploadDark = shallow(<FileUpload darkMode />);

    expect(fileUpload.props().className).toEqual('sc-crXcEl gVBqzM');
    expect(fileUploadDark.props().className).toEqual('sc-crXcEl MWJMz');
  });
});
