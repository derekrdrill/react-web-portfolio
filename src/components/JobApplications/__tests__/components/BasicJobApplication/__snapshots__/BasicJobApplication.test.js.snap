// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Job Application tests renders correctly 1`] = `
<Styled(Component)
  container={true}
  darkMode={true}
  spacing={2}
>
  <Styled(Component)
    container={true}
  >
    <ForwardRef(Grid)
      item={true}
      xs={12}
    >
      <Styled(Component)
        darkMode={true}
        variant="h4"
      >
        Application for Employment
      </Styled(Component)>
    </ForwardRef(Grid)>
  </Styled(Component)>
  <Styled(Component)
    darkMode={true}
    item={true}
    xs={12}
  >
    <ForwardRef(Typography)
      component="h5"
    >
      Upload supporting documents
    </ForwardRef(Typography)>
    <Styled(Component)
      container={true}
      darkMode={true}
    >
      <DynamicList
        addColor="forestgreen"
        removeColor="maroon"
      >
        <Styled(Component)
          darkMode={true}
          type="file"
        />
      </DynamicList>
    </Styled(Component)>
  </Styled(Component)>
  <DynamicFormInputs
    form={
      Object {
        "email": "",
        "job": "",
        "name": "",
        "phone": "",
        "summary": "",
      }
    }
    inputs={
      Array [
        Object {
          "fullWidth": true,
          "id": "name",
          "label": "Name",
          "md": 3,
          "sm": 6,
          "variant": "standard",
          "xs": 12,
        },
        Object {
          "fullWidth": true,
          "id": "email",
          "label": "Email",
          "md": 3,
          "sm": 6,
          "type": "email",
          "variant": "standard",
          "xs": 12,
        },
        Object {
          "fullWidth": true,
          "id": "phone",
          "label": "Phone",
          "md": 3,
          "sm": 6,
          "variant": "standard",
          "xs": 12,
        },
        Object {
          "fullWidth": true,
          "id": "job",
          "label": "What job are you applying for?",
          "md": 3,
          "sm": 6,
          "variant": "standard",
          "xs": 12,
        },
        Object {
          "fullWidth": true,
          "id": "summary",
          "label": "Write a short summary about yourself",
          "minRows": 3,
          "multiline": true,
          "variant": "standard",
          "xs": 12,
        },
      ]
    }
    setForm={[Function]}
  />
  <ForwardRef(Grid)
    container={true}
    justifyContent="flex-end"
  >
    <ForwardRef(Grid)
      item={true}
      lg={4}
      md={6}
      xs={12}
    >
      <Link
        replace={false}
        to="/app-complete"
      >
        <Styled(Component)
          color="info"
          fullWidth={true}
          size="large"
          variant="contained"
        >
          Submit Application
        </Styled(Component)>
      </Link>
    </ForwardRef(Grid)>
  </ForwardRef(Grid)>
</Styled(Component)>
`;
